aws/data_source_aws_ecr_authorization_token.go

```diff
@@ -0,0 +1,81 @@
+package aws
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsEcrAuthorizationToken() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsEcrAuthorizationTokenRead,
+
+		Schema: map[string]*schema.Schema{
+			"registry_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"authorization_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"proxy_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"expires_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"user_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"password": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEcrAuthorizationTokenRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ecrconn
+	params := &ecr.GetAuthorizationTokenInput{}
+	registryId := d.Get("registry_id")
+	if registryId != nil && len(registryId.(string)) > 0 {
+		params.RegistryIds = []*string{aws.String(registryId.(string))}
+	}
+	log.Printf("[DEBUG] Getting ECR authorization token")
+	out, err := conn.GetAuthorizationToken(params)
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("error getting ECR authorization token: %s", err)
+	}
+	log.Printf("[DEBUG] Received ECR AuthorizationData %v", out.AuthorizationData)
+	authorizationData := out.AuthorizationData[0]
+	authorizationToken := *authorizationData.AuthorizationToken
+	expiresAt := strconv.FormatInt(authorizationData.ExpiresAt.Unix(), 10)
```

Comment:
We [prefer RFC3339 format](https://www.terraform.io/docs/extend/schemas/schema-types.html#date-amp-time-data):

```go
expiresAt := aws.TimeValue(authorizationData.ExpiresAt).Format(time.RFC3339)
```

Reply:
fixed in 7bf9644d75ed3420a831aaff8bb98d053a2a0e26